lib/rubocop/cop/rspec/metadata_style.rb

```diff
@@ -0,0 +1,217 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module RSpec
+      # Use consistent metadata style.
+      #
+      # This cop does not support autocorrection in the case of
+      # `EnforcedStyle: hash` where the trailing metadata type is ambiguous.
+      # (e.g. `describe 'Something', :a, b`)
+      #
+      # @example EnforcedStyle: symbol (default)
+      #   # bad
+      #   describe 'Something', a: true
+      #
+      #   # good
+      #   describe 'Something', :a
+      #
+      # @example EnforcedStyle: hash
+      #   # bad
+      #   describe 'Something', :a
+      #
+      #   # good
+      #   describe 'Something', a: true
+      class MetadataStyle < Base # rubocop:disable Metrics/ClassLength
+        extend AutoCorrector
+
+        include ConfigurableEnforcedStyle
+        include Metadata
+        include RangeHelp
+
+        # @!method extract_metadata_hash(node)
+        def_node_matcher :extract_metadata_hash, <<~PATTERN
+          (send _ _ _ ... $hash)
+        PATTERN
+
+        # @!method match_boolean_metadata_pair?(node)
+        def_node_matcher :match_boolean_metadata_pair?, <<~PATTERN
+          (pair sym true)
+        PATTERN
+
+        # @!method match_ambiguous_trailing_metadata?(node)
+        def_node_matcher :match_ambiguous_trailing_metadata?, <<~PATTERN
+          (send _ _ _ ... !{hash sym})
+        PATTERN
+
+        def on_metadata(symbols, hash)
+          on_empty_metadata_hash(hash) if hash&.pairs&.empty?
+
+          hash&.pairs&.each do |pair|
+            on_metadata_pair(pair)
+          end
+
+          symbols.each do |symbol|
+            on_metadata_symbol(symbol)
+          end
```

Comment:
Redundant navigation can be eliminated by changing and adjusting the processing flow.

```suggestion
          symbols.each do |symbol|
            on_metadata_symbol(symbol)
          end

          return unless hash

          on_empty_metadata_hash(hash) if hash.pairs.empty?

          hash.pairs.each do |pair|
            on_metadata_pair(pair)
          end
```

          on_empty_metadata_hash(hash) if hash.pairs.empty?

          hash.pairs.each do |pair|
            on_metadata_pair(pair)
          end
```